pkg/config/resource.go

```diff
@@ -184,6 +185,21 @@ func (s *Sensitive) AddFieldPath(tf, xp string) {
 	s.fieldPaths[tf] = xp
 }
 
+// OperationTimeouts allows configuring resource operation timeouts:
+// https://www.terraform.io/language/resources/syntax#operation-timeouts
+// Please note that, not all resources support configuring timeouts.
+type OperationTimeouts struct {
+	Read   *time.Duration
+	Create *time.Duration
+	Update *time.Duration
+	Delete *time.Duration
```

Comment:
```suggestion
	Read   time.Duration
	Create time.Duration
	Update time.Duration
	Delete time.Duration
```

I am guessing that you had to use pointers to check whether it's given or not but there is a function `(t Time) IsZero() bool` we can use to do that. If that's the only reason, we could get rid of `TimeDuration(d time.Duration) *time.Duration` and work with value types.

Reply:
Yes, that was the reason and I was thinking that setting timeout to 0 could be a valid scenario, e.g. no timeout. But digging the terraform sdk code, I noticed that there is no such an implementation and probably no timeout does not make sense at all.

So, I would revert pointers but to check zero value I can not use `IsZero()` because it is on `time.Time` not `time.Duration`. Am I missing something?

Otherwise, I will go with the following:
```
	if t := fp.Config.OperationTimeouts.Read.String(); t != "0s" {
		timeouts["read"] = t
	}
```

Reply:
Oh yes, I missed that `IsZero` wouldn't work. I think you can check like `fp.Config.OperationTimeouts.Read != 0` since `type Duration int64`